demo/playground/README.md

```diff
@@ -0,0 +1,124 @@
+
+This directory contains scripts to run several Aca-Py agents in various configurations for demonstration, development and testing scenarios. The agents are using Postgres (15) database storage (`askar`, `DatabasePerWallet`) and are running without security (`--admin-insecure-mode`).
+
+The inspiration for this playground was testing mediation and the differences between single-tenant and multi-tenant modes. These scripts allowed the developer to stand up 3 single-tenant agents and 1 multi-tenant agent and run various scenarios (in other scripts - not included). Running in `--admin-insecure-mode` simplifies creating tenants in multi-tenant mode and eliminates the need for adding headers for calls in single-tenant mode.
+
+- faber-agent
+- alice-agent
+- acme-agent
+- multi-agent (the multi-tenant agent)
+
+By default, all the agents share the same Postgres Database Service (version 15) and all use [Ngrok](https://ngrok.com) for publicly accessible URLs.
+
+
+### Dependencies
+Docker Compose version v2.17.2
+
+## Agent Configuration
+
+There are two extremely simple configurations provided:
+
+- [`singletenant-auto-accept.yml`](./configs/singletenant-auto-accept.yml)
+- [`multitenant-auto-accept.yml`](./configs/multitenant-auto-accept.yml)
+
+These configuration files are provided to the Aca-Py start command via the `AGENT_ARG_FILE` environment variable. See [`.env`](./.env.sample) and [`start.sh`](./start.sh).
+
+Additional (and more complicated) configurations coming soon... maybe.
+
+### Dockerfile and start.sh
+
+[`Dockerfile.acapy`](./Dockerfile.acapy) assembles the image to run. Currently based on [Aries Cloudagent Python 0.8.1](ghcr.io/hyperledger/aries-cloudagent-python:py3.9-indy-1.16.0-0.8.1), we need [jq](https://stedolan.github.io/jq/) to setup (or not) the ngrok tunnel and execute the Aca-py start command - see [`start.sh`](./start.sh). You may note that the start command is very sparse, additional configuration is done via environment variables in the [docker compose file](./docker-compose.yml).
+
+### ngrok
+Note that ngrok allows 2 tunnels per instance with an unpaid account. We have broken up the 4 default services into 2 ngrok services and tunnel configurations. If you need to alter port numbers for your agent services, you will have to update the ngrok tunnel files.
+
+- [ngrok-faber-alice](./ngrok-faber-alice.yml)
+- [ngrok-acme-multi](./ngrok-acme-multi.yml)
+
+If you have a paid account, you can set the `NGROK_AUTHTOKEN` environment variable. See below.
+
+### .env
+Additional configuration (ie. port numbers for the services, `NGROK_AUTHTOKEN`, ...) are done in the [`.env`](./.env.sample) file. Change as needed and ensure ngrok configuration matches.
+
+```shell
+cp .env.sample .env
+```
+
+
+## Running the Playground
+
+To run the agents in this repo, open a command shell in this directory and run:
+
+- to build the containers:
+
+```bash
+docker compose build
+```
+
+- to run the agents:
+
+```bash
+docker compose up
```

Comment:
On starting, I get some errors. Do I care?

```
playground-multi-agent-1        | 2023-05-12 20:36:45,292 aries_cloudagent.core.profile INFO Create profile manager: askar
playground-alice-agent-1        | 2023-05-12 20:36:45,299 aries_cloudagent.core.profile INFO Create profile manager: askar
playground-wallet-db-1          | 2023-05-12 20:36:45.314 UTC [80] FATAL:  database "multi_wallet" does not exist
playground-acme-agent-1         | 2023-05-12 20:36:45,320 aries_cloudagent.core.profile INFO Create profile manager: askar
playground-wallet-db-1          | 2023-05-12 20:36:45.323 UTC [81] FATAL:  database "alice_wallet" does not exist
playground-wallet-db-1          | 2023-05-12 20:36:45.324 UTC [82] FATAL:  database "multi_wallet" does not exist
playground-wallet-db-1          | 2023-05-12 20:36:45.333 UTC [83] FATAL:  database "alice_wallet" does not exist
playground-wallet-db-1          | 2023-05-12 20:36:45.349 UTC [86] FATAL:  database "acme_wallet" does not exist
playground-wallet-db-1          | 2023-05-12 20:36:45.360 UTC [87] FATAL:  database "acme_wallet" does not exist
```

Reply:
Not an issue, the databases will get created.